android/replapp/cpp/ReplProxy.cpp

```diff
@@ -0,0 +1,207 @@
+#include "hermes/Platform/Intl/PlatformIntl.h"
+
+#include <fbjni/fbjni.h>
+#include "hermes/VM/Callable.h"
+
+#define C_STRING(x) #x
+
+using namespace ::facebook;
+using namespace ::hermes;
+
+class ReplProxy : public facebook::jni::HybridClass<ReplProxy> {
+
+    std::shared_ptr<vm::Runtime> createAndGetRuntime() {
+        vm::RuntimeConfig config = vm::RuntimeConfig::Builder()
+                .withGCConfig(
+                        vm::GCConfig::Builder()
+                                .withInitHeapSize(32 << 20)
+                                .withMaxHeapSize(512 << 20)
+                                .withSanitizeConfig(vm::GCSanitizeConfig::Builder()
+                                                            .withSanitizeRate(0.0)
+                                                            .withRandomSeed(-1)
+                                                            .build())
+                                .withShouldRecordStats(false)
+                                .withCallback([this](vm::GCEventKind gcEventKind,
+                                                     const char *extraInfo) {
+                                    GCCallback(gcEventKind, extraInfo);
+                                })
+                                .build())
+                .withVMExperimentFlags(vm::RuntimeConfig::getDefaultVMExperimentFlags())
+                .withES6Promise(vm::RuntimeConfig::getDefaultES6Promise())
+                .withES6Proxy(vm::RuntimeConfig::getDefaultES6Proxy())
+                .withES6Symbol(vm::RuntimeConfig::getDefaultES6Symbol())
+                .withES6Intl(true)
+                .withEnableHermesInternal(true)
+                .withEnableHermesInternalTestMethods(true)
+                .withAllowFunctionToStringWithRuntimeSource(false)
+                .build();
+
+        return vm::Runtime::create(config);
+    }
+
+    vm::HermesValue getEvaluateLineFn(const std::shared_ptr<vm::Runtime>& runtime) {
+
+        vm::GCScope gcScope(runtime.get());
+        runtime->getHeap().runtimeWillExecute();
+
+        vm::Handle <vm::JSObject> global = runtime->getGlobal();
+
+        // evaluate-line.js will color the output appropriately based on the available of this property in the global object.
+        vm::HermesValue terminalTypePropName = vm::HermesValue::encodeStringValue(vm::StringPrimitive::createNoThrow(runtime.get(), "_replterminaltype").get());
+        vm::HermesValue terminalTypePropVal = vm::HermesValue::encodeStringValue(vm::StringPrimitive::createNoThrow(runtime.get(), "android").get());
```

Comment:
You can't pass `vm::HermesValue` around across `putComputed_RJS` which might allocate (hence result in GC and move objects). Instead, `vm::StringPrimitive::createNoThrow` already returns `Handle` and you can directly pass those `Handle` without the need to wrap them in `makeHandle` at callsite.

Reply:
Thanks ! Done.